trio/tests/test_file_io.py

```diff
@@ -0,0 +1,169 @@
+        getattr(async_file, 'unsupported_attr')
+
+
+def test_sync_attrs_forwarded(async_file, wrapped):
+    for attr_name in _FILE_SYNC_ATTRS:
+        if attr_name not in dir(async_file):
+            continue
+
+        assert getattr(async_file, attr_name) == getattr(wrapped, attr_name)
+
+
+def test_sync_attrs_match_wrapper(async_file, wrapped):
+    for attr_name in _FILE_SYNC_ATTRS:
+        if attr_name in dir(async_file):
+            continue
+
+        with pytest.raises(AttributeError):
+            getattr(async_file, attr_name)
+
+        with pytest.raises(AttributeError):
+            getattr(wrapped, attr_name)
+
+
+def test_async_methods_generated_once(async_file):
+    for meth_name in _FILE_ASYNC_METHODS:
+        if meth_name not in dir(async_file):
+            continue
+
+        assert getattr(async_file, meth_name) == getattr(async_file, meth_name)
+
+
+def test_async_methods_signature(async_file):
+    # use read as a representative of all async methods
+    assert async_file.read.__name__ == 'read'
+    assert async_file.read.__qualname__ == 'AsyncIOWrapper.read'
+
+    assert 'io.StringIO.read' in async_file.read.__doc__
+
+
+async def test_async_methods_wrap(async_file, wrapped):
+    for meth_name in _FILE_ASYNC_METHODS:
+        if meth_name not in dir(async_file):
+            continue
+
+        meth = getattr(async_file, meth_name)
+        wrapped_meth = getattr(wrapped, meth_name)
+
+        value = await meth(sentinel.argument, keyword=sentinel.keyword)
+
+        wrapped_meth.assert_called_once_with(sentinel.argument, keyword=sentinel.keyword)
+        assert value == wrapped_meth()
+
+        wrapped.reset_mock()
+
+
+async def test_async_methods_match_wrapper(async_file, wrapped):
+    for meth_name in _FILE_ASYNC_METHODS:
+        if meth_name in dir(async_file):
+            continue
+
+        with pytest.raises(AttributeError):
+            getattr(async_file, meth_name)
+
+        with pytest.raises(AttributeError):
+            getattr(wrapped, meth_name)
+
+
+async def test_open(path):
+    f = await trio.open_file(path, 'w')
+
+    assert isinstance(f, trio.AsyncIOWrapper)
+
+    await f.close()
+
+
+async def test_open_context_manager(path):
+    async with await trio.open_file(path, 'w') as f:
+        assert isinstance(f, trio.AsyncIOWrapper)
+        assert not f.closed
+
+    assert f.closed
+
+
+async def test_async_iter(async_file):
+    async_file._wrapped = io.StringIO('test\nfoo\nbar')
+    expected = iter(list(async_file.wrapped))
+    async_file.wrapped.seek(0)
+
+    async for actual in async_file:
+        assert actual == next(expected)
+
+    with pytest.raises(StopIteration):
+        next(expected)
+
+
+async def test_close_cancelled(path):
+    with _core.open_cancel_scope() as cscope:
+        f = await trio.open_file(path, 'w')
+        cscope.cancel()
+
+        with pytest.raises(_core.Cancelled):
+            await f.write('a')
+
+        await f.close()
```

Comment:
This line should also raise `Cancelled`, so we might as well add a `with pytest.raises(...)` to assert that.